dnf/conf/substitutions.py

```diff
@@ -60,7 +60,7 @@ def update_from_etc(self, installroot, varsdir=("/etc/yum/vars/", "/etc/dnf/vars
                             val = fp.readline()
                         if val and val[-1] == '\n':
                             val = val[:-1]
-                    except (OSError, IOError):
+                    except (OSError, IOError, UnicodeDecodeError):
```

Comment:
The patch resolves the issue but the problem is that it does not enforce or inform user about problem on file system. I also checked that except (OSError, IOError) is here from beginning therefore we should be careful. What about to create a warning that some files cannot be correctly opened or processed to bring an attention to the user?